src/Common/src/Interop/Windows/mincore/Interop.GetLongPathNameW.cs

```diff
@@ -8,10 +9,30 @@ partial class Interop
 {
     partial class mincore
     {
+        /// <summary>
+        /// WARNING: This overload does not implicitly handle long paths.
+        /// </summary>
         [DllImport(Libraries.CoreFile_L1, SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
         internal unsafe static extern int GetLongPathNameW(char* path, char* longPathBuffer, int bufferLength);
 
-        [DllImport(Libraries.CoreFile_L1, SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
-        internal static extern int GetLongPathNameW(string path, [Out]StringBuilder longPathBuffer, int bufferLength);
+        [DllImport(Libraries.CoreFile_L1, EntryPoint = "GetLongPathNameW", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
+        private static extern int GetLongPathNameWPrivate(string path, [Out]StringBuilder longPathBuffer, int bufferLength);
+
+        internal static int GetLongPathNameW(string path, [Out]StringBuilder longPathBuffer, int bufferLength)
+        {
+            bool wasExtended = PathInternal.IsExtended(path);
+            if (!wasExtended)
+            {
+                path = PathInternal.AddExtendedPathPrefixForLongPaths(path);
```

Comment:
In the previous review, you said we couldn't change AddExtendedPathPrefixForLongPaths to AddExtendedPathPrefixIfNecessary because it would conflict with something in this next PR.  I'm not seeing it.  What's the conflict?